### PR TITLE
Move WattTime Client to live within WattTime Datasource

### LINKED ISSUE #218: [Feature Contribution]: Move WattTime Client to live within WattTime Datasource
### What happened?

WattTime Client was created as a separate project from the WattTime DataSource project. As we have moved forward with the SDK, the WattTime data source is tightly integrated into the Client and we now believe the Client should live fully within the WattTime data source. This feature moves the client and all accompanying tests. This is also in line with the current ElectricityMaps implementation that is up for review as well.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Feature Commitment

- [X] I commit to contributing this feature as a PR and working with the GSF to merge this feature into the Carbon Aware SDK.

### PR DESCRIPTION
**Pull request by JenMadiedo**
Issue Number: #218

# DEPENDENT PR
> **Warning**
> This PR depends on commits in other PRs
> **DO NOT** merge this PR until https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/203 is merged and this branch is updated against the latest `dev` branch
## Summary
Move the WattTime Client out of its own project and to live within the WattTime DataSource project. No functional code changes, only movement of files/updating project references

## Changes

- Move WattTime Client to WattTime DataSource project
- Move WattTime Client tests to own test folder in WattTime DataSource project
- Modify references and .sln file to reflect move

## Checklist

- [X] Local Tests Passing?
- [X] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [ ] Are there any API Changes? If yes, please describe below.
- [ ] This is not a breaking change. If it is, please describe it below.

This PR Closes #218

### CODE REVIEW

File: src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/Client/InternalsVisibleTo.cs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+using System.Runtime.CompilerServices;
```
Comment (juzuluag) [inline src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/Client/InternalsVisibleTo.cs:1]: I would recommend to put this into the `csproj` file. No need for a new file.
See `GSF.CarbonAware.csproj` as an example

File: src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/Client/WattTimeClientTests.cs [hunk 2]
```diff
@@ -106,7 +106,7 @@ public async Task GetDataAsync_DeserializesExpectedResponse()
 
         var data = await client.GetDataAsync("balauth", new DateTimeOffset(2022, 4, 22, 0, 0, 0, TimeSpan.Zero), new DateTimeOffset(2022, 4, 22, 0, 0, 0, TimeSpan.Zero));
 
-        Assert.IsTrue(data.Count() > 0);
```
Comment (juzuluag) [inline src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/Client/WattTimeClientTests.cs:109]: Any reason why these changes with this PR? Looks like they are not related with the PR goal.

**Review by juzuluag — CHANGES_REQUESTED** [review]
Overall looks great. Few comments.

File: src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/Configuration/ServiceCollectionExtensionTests.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
+using CarbonAware.Configuration;
```
[the remark this answers is not in the document]
Comment (JenMadiedo) [reply src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/Configuration/ServiceCollectionExtensionTests.cs:1]: 2 new files with existing tests. Split these tests off from the main WattTime tests to line up with the EM data source test structure

File: src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/Client/InternalsVisibleTo.cs (hunk 1 above)
Comment (JenMadiedo) [reply src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/Client/InternalsVisibleTo.cs:1]: will do!

File: src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/Client/WattTimeClientTests.cs (hunk 2 above)
Comment (JenMadiedo) [reply src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/Client/WattTimeClientTests.cs]: Oh these were unintentional, I ran a formatting script that cleaned this up unintentionally. I'll revert

File: src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/CarbonAware.DataSources.WattTime.csproj [hunk 4]
```diff
@@ -9,7 +9,27 @@
...
+    </AssemblyAttribute>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="Properties\" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Net.Http.Headers" Version="2.2.8" />
+    <PackageReference Include="WireMock.Net" Version="1.4.43"/>
```
Comment (bderusha) [inline src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/CarbonAware.DataSources.WattTime.csproj:29]: Any reason why this and `Http.Headers` lives here versus in the `/mock` `/test` projects?

**Review by bderusha — APPROVED** [review]
One question about project dependencies, otherwise :shipit: 

Nice work!

**Review by juzuluag — APPROVED** [review]

File: src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/CarbonAware.DataSources.WattTime.csproj (hunk 4 above)
Comment (JenMadiedo) [reply src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/CarbonAware.DataSources.WattTime.csproj:29]: I think it was just originally written like that. I tried to remove but the dependency chain is complicated and I haven't gotten the right combo of packages, so I'll table for another task

**Comment by JenMadiedo** [comment]
@vaughanknight @Willmish this is ready for review

**Review by bderusha — APPROVED** [review]

**Review by vaughanknight — APPROVED** [review]
Look good.  Thank you.
